### PR TITLE
request/nonce: simplify and prevent two consecutive mutex lock calls 

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Requires #1486 

Attempts to improve flow when deriving a nonce. 

* Also changes function signature so the nonce package doesn't need to handle setting different values, this can be defined caller side so that anything can be put in, so we don't need to do anything in the future if milliseconds, microseconds, 1 etc need to be added later. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — CHANGES_REQUESTED** [review]
The double lock only appears to occur on the first time a request nonce is used. Which is once per exchange

File: exchanges/request/request_test.go [hunk 1]
```diff
@@ -494,37 +496,35 @@ func TestDoRequest_NotRetryable(t *testing.T) {
...
-		new(http.Client),
-		WithLimiter(&globalshell))
-	if err != nil {
-		t.Fatal(err)
-	}
-	if m1, m2 := r.GetNonceMilli(), r.GetNonceMilli(); m1 == m2 {
-		log.Fatal(unexpected)
+// 40532461	       30.29 ns/op	       0 B/op	       0 allocs/op (prev)
+// 41983143	       28.44 ns/op	       0 B/op	       0 allocs/op
+func BenchmarkGetNonce(b *testing.B) {
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/request/request_test.go:522]: Thanks for making it more performant than `master`:
```
BenchmarkGetNonce-10    	14245105	        83.73 ns/op
```

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for improving things for my 💩 💻

**Review by thrasher- — APPROVED** [review]
tACK, nice simplification🌞

**Pull request merged**